pkg/innerring/processors/netmap/nodevalidation/external/calls.go

```diff
@@ -0,0 +1,144 @@
+package external
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neofs-sdk-go/netmap"
+)
+
+// Signature represents a digital signature of a signed message.
+type Signature struct {
+	Sign []byte `json:"sign"`
+}
+
+// RequestBody represents the body of a request sent
+// to an external validator.
+type RequestBody struct {
+	NodeInfo netmap.NodeInfo `json:"node_info"`
+	Nonce    uint32          `json:"nonce"`
+}
+
+// SignedRequest represents a request containing node information
+// and its corresponding digital signature.
+type SignedRequest struct {
+	Body      json.RawMessage `json:"body"`
+	Signature *Signature      `json:"signature"`
+}
+
+// ResponseBody represents the body of the response from an external validator.
+type ResponseBody struct {
+	Verified bool   `json:"verified"`
+	Details  string `json:"details"`
+	Nonce    uint32 `json:"nonce"`
+}
+
+// SignedResponse represents the result of a verification
+// process from an external validator.
+// Verified indicates whether the verification was successful.
+type SignedResponse struct {
```

Comment:
`SignedRequest` can be reused here, btw. Just body/signature. Then bodies are different.